fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingScrubMissing.java

```diff
@@ -217,16 +217,10 @@ public IndexingScrubMissing(@Nonnull final IndexingCommon common,
                     final byte[] keyBytes = maintainer.getIndexSubspace().pack(valueKey);
                     return maintainer.state.transaction.get(keyBytes).thenApply(indexVal -> indexVal == null ? valueKey : null);
                 })
+                .filter(Objects::nonNull)
```

Comment:
This will never filter anything (sorry, this was my fault), as the `.filter()` is on the stream that is producing futures that are looking for the missing keys, so it is filtering on the futures, not the result of the futures.

Reply:
NP. Fixing...